TaskMarsCompetition/TestMarsCompetition/Utilities/CommonHooks.cs

```diff
@@ -0,0 +1,182 @@
+﻿
+using AventStack.ExtentReports;
+using AventStack.ExtentReports.Reporter;
+using NUnit.Framework.Interfaces;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Chrome;
+using TestMarsCompetition.Utilities;
+using TestMarsCompetition.Context;
+
+using TestMarsCompetition.Page;
+using TestMarsCompetition.ModelEducation;
+using TestMarsCompetition.ModelLogin;
+
+
+namespace TestMarsCompetition.Utilities
+{
+    
+    public class CommonHooks
+    {
+        public static AventStack.ExtentReports.ExtentReports extent;
+        public static ExtentTest test;
+        private EducationPage educationPage;
+        private CertificatePage certificatePage;
+        private TestDataLogin loginData;
+
+        private Login login;
+
+        private TestData testData;
+        string testName = TestContext.CurrentContext.Test.Name;
+
+        public CommonHooks()
+        {
+            extent = new AventStack.ExtentReports.ExtentReports();
+            login = new Login();
+            educationPage = new EducationPage();
+            certificatePage = new CertificatePage();
+            
+        }
+
+
+        //report file
+        [OneTimeSetUp]
+        public void Setup()
+
+        {
+            //Create directory for the reporting
+            string workingDirectory = Environment.CurrentDirectory;
+            string projectDirectory = Directory.GetParent(workingDirectory).Parent.Parent.FullName;
+
+           
+            String reportPath = projectDirectory + "//index.html";
+            var htmlReporter = new ExtentHtmlReporter(reportPath);
+
+            extent.AttachReporter(htmlReporter);
+            extent.AddSystemInfo("Host Name", "Local host");
+            extent.AddSystemInfo("Environment", "QA");
+            extent.AddSystemInfo("Username", "R");
+
+        }
+        [SetUp]
+        public  void StartUp()
+        {
+            
+            test = extent.CreateTest(testName);
+            WebdriverManager.InitializeDriver();
+            loginData = JsonReaderlogin.ReadTestData("Utilities/TestDataLogin.json");
+             login.loginPage(loginData.email, loginData.password);
+            
+
+            if (testName.Contains("Education", StringComparison.OrdinalIgnoreCase))
+            {
+                educationPage.GoToTab();
+                educationPage.DeleteAllElements();
+            }
+            else if (testName.Contains("Certificate", StringComparison.OrdinalIgnoreCase))
+            {
+                certificatePage.GoToTab();
+                certificatePage.DeleteAllElements();
```

Comment:
so this is still not correct, you need to did all data  once before test suit run so this should be in a hook that run before any tests, ideally        [OneTimeSetUp] 

then after the each test, the test should clean up its on data